openstack_controller/datadog_checks/openstack_controller/components/network.py

```diff
@@ -0,0 +1,115 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+
+from datadog_checks.base.utils.discovery import Discovery
+from datadog_checks.openstack_controller.components.component import Component
+from datadog_checks.openstack_controller.config import normalize_discover_config_include
+from datadog_checks.openstack_controller.metrics import (
+    NEUTRON_AGENTS_COUNT,
+    NEUTRON_AGENTS_METRICS,
+    NEUTRON_AGENTS_METRICS_PREFIX,
+    NEUTRON_AGENTS_TAGS,
+    NEUTRON_NETWORK_COUNT,
+    NEUTRON_NETWORK_METRICS,
+    NEUTRON_NETWORK_METRICS_PREFIX,
+    NEUTRON_NETWORK_TAGS,
+    NEUTRON_QUOTA_METRICS,
+    NEUTRON_QUOTA_METRICS_PREFIX,
+    NEUTRON_QUOTA_TAGS,
+    NEUTRON_RESPONSE_TIME,
+    NEUTRON_SERVICE_CHECK,
+    get_metrics_and_tags,
+)
+
+
+class Network(Component):
+    ID = Component.Id.NETWORK
+    TYPES = Component.Types.NETWORK
+    SERVICE_CHECK = NEUTRON_SERVICE_CHECK
+
+    def __init__(self, check):
+        super(Network, self).__init__(check)
+
+    @Component.register_global_metrics(ID)
+    @Component.http_error(report_service_check=True)
+    def _report_response_time(self, global_components_config, tags):
```

Comment:
What do you think about abstracting the `_report_response_time` into the Component class?

Reply:
Yep. I have this and some more ideas about Component class, but I prefer leave them for a future PR.